src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-config-modal/rgw-config-modal.component.ts

```diff
@@ -47,6 +57,50 @@ export class RgwConfigModalComponent implements OnInit {
     this.kmsProviders = this.rgwEncryptionModal.kmsProviders;
     this.authMethods = this.rgwEncryptionModal.authMethods;
     this.secretEngines = this.rgwEncryptionModal.secretEngines;
+    if (this.editing && this.selectedEncryptionConfigValues) {
+      this.configForm.get('address').setValue(this.selectedEncryptionConfigValues['addr']);
+      if (this.selectedEncryptionConfigValues['encryption_type'] === 'SSE-KMS') {
+        this.configForm.get('encryptionType').setValue(ENCRYPTION_TYPES.SSE_KMS);
+      } else if (this.selectedEncryptionConfigValues['encryption_type'] === 'SSE-S3') {
+        this.configForm.get('encryptionType').setValue(ENCRYPTION_TYPES.SSE_S3);
+      }
+      this.configForm.get('kms_provider').setValue(this.selectedEncryptionConfigValues['backend']);
+      this.configForm.get('kms_provider').disable();
+      this.configForm.get('auth_method').setValue(this.selectedEncryptionConfigValues['auth']);
+      this.configForm
+        .get('secret_engine')
+        .setValue(this.selectedEncryptionConfigValues['secret_engine']);
+      this.configForm.get('secret_path').setValue(this.selectedEncryptionConfigValues['prefix']);
+      this.configForm.get('namespace').setValue(this.selectedEncryptionConfigValues['namespace']);
```

Comment:
i feel like this part of the code can be heavily simplified and make it more readable if we start using `patchValue` from the angular.